test/suites/integration/extended_resources_test.go

```diff
@@ -158,6 +163,54 @@ var _ = Describe("Extended Resources", func() {
 		env.ExpectCreatedNodeCount("==", 1)
 		env.EventuallyExpectCreatedNodesInitialized()
 	})
+	It("should provision nodes for a deployment that requests amd.com/gpu", func() {
+		ExpectAMDDevicePluginCreated()
+		DeferCleanup(func() {
+			ExpectAMDDevicePluginDeleted()
+		})
+
+		content, err := os.ReadFile("testdata/amd_driver_input.golden")
+		Expect(err).ToNot(HaveOccurred())
+		provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{
+			SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+		},
+			UserData: aws.String(string(content)),
+		})
+		provisioner := test.Provisioner(test.ProvisionerOptions{
+			ProviderRef: &v1alpha5.ProviderRef{Name: provider.Name},
+			Requirements: []v1.NodeSelectorRequirement{
+				{
+					Key:      v1alpha1.LabelInstanceCategory,
+					Operator: v1.NodeSelectorOpExists,
+				},
+			},
+		})
+		numPods := 1
+		dep := test.Deployment(test.DeploymentOptions{
+			Replicas: int32(numPods),
+			PodOptions: test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": "large-app"},
+				},
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						"amd.com/gpu": resource.MustParse("1"),
+					},
+					Limits: v1.ResourceList{
+						"amd.com/gpu": resource.MustParse("1"),
+					},
+				},
+			},
+		})
+		selector := labels.SelectorFromSet(dep.Spec.Selector.MatchLabels)
+		env.ExpectCreated(provisioner, provider, dep)
+		EventuallyWithOffset(1, func(g Gomega) {
```

Comment:
Why not use `ExpectPodsHealthy`?

Reply:
The driver installation requires the Timeout value to be extended. The `ExpectPodsHealthy` does not allow for that change.